build-support/docker/travis_ci/Dockerfile

```diff
@@ -4,19 +4,6 @@
 # Use our custom Centos6 image for binary compatibility with old linux distros.
 FROM pantsbuild/centos6:latest
 
-# TODO(7064) These changes belong in centos6/Dockerfile, because we always want that image to
-# have Py3 available and also avoid the cost of rebuilding Python 3 every time in CI. However,
-# this requires publishing an updated image to Docker hub, so we copy the code here until then.
-ARG PYTHON_3_VERSION=3.6.8
-RUN yum install sqlite-devel -y
-ENV PYENV_ROOT /pyenv-docker-build
-RUN mkdir ${PYENV_ROOT}
-RUN git clone https://github.com/pyenv/pyenv ${PYENV_ROOT}
-RUN /usr/bin/scl enable devtoolset-7 -- bash -c '\
-    ${PYENV_ROOT}/bin/pyenv install ${PYTHON_3_VERSION} \
-    && ${PYENV_ROOT}/bin/pyenv global ${PYTHON_3_VERSION}'
-ENV PATH "${PYENV_ROOT}/shims:${PATH}"
-
```

Comment:
Does this end up making the first CI stage faster by 3 minutes in real time as well? Or are we still paying that cost when installing python 3 for the py 3 linux pex shard in stage 1, and the win here is just in overall time across all shards?

Reply:
Unfortunately win is only in overall CI time, not in wall time. We still have to build Py36 for `Linux Build Engine Py36` and `Linux Build Wheels UCS4`, and soon the two wheel building shards for Py3. 

Which is why #7064 will be so great! And your PR will also help with Py37 to save time with our cron job, once I figure out #7261.